modules/apps/29-fee/types/msgs.go

```diff
@@ -41,3 +44,135 @@ func (msg MsgRegisterCounterpartyAddress) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{signer}
 }
+
+// NewMsgPayPacketFee creates a new instance of MsgPayPacketFee
+func NewMsgPayPacketFee(fee Fee, sourcePortId, sourceChannelId, signer string, relayers []string) *MsgPayPacketFee {
+	return &MsgPayPacketFee{
+		Fee:             fee,
+		SourcePortId:    sourcePortId,
+		SourceChannelId: sourceChannelId,
+		Signer:          signer,
+		Relayers:        relayers,
+	}
+}
+
+// ValidateBasic performs a basic check of the MsgPayPacketFee fields
+func (msg MsgPayPacketFee) ValidateBasic() error {
+	// validate channelId
+	err := host.ChannelIdentifierValidator(msg.SourceChannelId)
+	if err != nil {
+		return err
+	}
+
+	// validate portId
+	err = host.PortIdentifierValidator(msg.SourcePortId)
+	if err != nil {
+		return err
+	}
+
+	// signer check
+	_, err = sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return sdkerrors.Wrap(err, "failed to convert msg.Signer into sdk.AccAddress")
+	}
+
+	// enforce relayer is nil
+	if msg.Relayers != nil {
+		return ErrRelayersNotNil
+	}
+
+	// ensure fee's are valid amounts
+	if !msg.Fee.AckFee.IsValid() {
+		return sdkerrors.ErrInvalidCoins
+	}
+
+	if !msg.Fee.ReceiveFee.IsValid() {
+		return sdkerrors.ErrInvalidCoins
+	}
+
+	if !msg.Fee.TimeoutFee.IsValid() {
+		return sdkerrors.ErrInvalidCoins
+	}
```

Comment:
Should we check that one of the coins is non-zero?

Reply:
agreed this is needed for sync and async messages

Reply:
good catch. I've updated this & added tests.